### PR TITLE
recipes-test: Add new utilities to 'initramfs-test-image.bb'

### PR DESCRIPTION
**Pull request by bhupesh-sharma**
Add following utilities to 'initramfs-test-image.bb':

-> For kexec/kdump analysis:
   - crash
   - makedumpfile
   - kexec
-> iozone3: for filesystem performance analysis.
-> cryptsetup: for crypto benchmarks.
-> util-linux: for 'gdisk', 'taskset', etc.
-> coreutils: mainly for 'dd' (as busybox 'dd' does not show performance
              figures).

Addition of the above tools results in an approximate increase of
6M in the resulting rootfs size (cpio.gz archive)

Before:
-------
$ ls -lah initramfs-test-image-qcom-armv8a-20210521045425.rootfs.cpio.gz
19M

After:
------
$ ls -lah initramfs-test-image-qcom-armv8a-20210521070020.rootfs.cpio.gz
25M

Signed-off-by: Bhupesh Sharma <bhupesh.sharma@linaro.org>

### CODE REVIEW

File: recipes-test/images/initramfs-test-image.bb [hunk 1]
```diff
@@ -5,6 +5,9 @@ export IMAGE_BASENAME = "initramfs-test-image"
 
 PACKAGE_INSTALL += " \
     bluez5 \
+    coreutils \
+    crash \
+    cryptsetup \
```
Comment (ndechesne) [inline recipes-test/images/initramfs-test-image.bb:10]: cryptsetup is already in the image, see 78e2dacd0ebcf6719ab122ed65975351e3bd87db.

in the main bb file you should only add packages which come from openembedded-core, if you include packages from other layers, we need to use the ./dynamic-layers/ subfolder instead. meta-qcom *must* build with just oe-core enabled.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend [hunk 2]
```diff
@@ -1,4 +1,9 @@
 PACKAGE_INSTALL += " \
+    coreutils \
```
Comment (ndechesne) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:2]: the bbappend in dynamic-layers is used when meta-oe is added to the user's build. This is to ensure that if someone does not build with meta-oe, then the build won't fail. 

so you need to add packages from meta-oe in this file, but packages from oe-core can be in the main bb recipe, they don't need to be here.

Comment (bhupesh-sharma) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:2]: Ok, new version v2 has been posted.

File: recipes-test/images/initramfs-test-image.bb [hunk 3]
```diff
@@ -14,14 +17,18 @@ PACKAGE_INSTALL += " \
     e2fsprogs-tune2fs \
     ethtool \
     gptfdisk \
+    iozone3 \
     iw \
+    kexec \
```
[the remark this answers is not in the document]
Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb]: Ok, new version v2 has been posted.

File: recipes-test/images/initramfs-test-image.bb (hunk 1 above)
Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb]: Ok, new version v2 has been posted.

File: recipes-test/images/initramfs-test-image.bb [hunk 4]
```diff
@@ -17,10 +17,16 @@ PACKAGE_INSTALL += " \
     iw \
     lava-test-shell \
     libdrm-tests \
+    lsof \
+    ncurses \
+    ncurses-terminfo \
```
Comment (lumag) [inline recipes-test/images/initramfs-test-image.bb:22]: do you need full terminfo?

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend [hunk 5]
```diff
@@ -1,4 +1,14 @@
...
+    crash \
     cryptsetup \
     devmem2 \
+    iozone3 \
+    kexec \
+    libgpiod \
```
Comment (lumag) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:8]: I think libgpiod would be picked up by libgpiod-tests, wouldn't it?

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend [hunk 6]
```diff
@@ -1,4 +1,14 @@
...
+    libgpiod-tools \
+    makedumpfile \
+    mbw \
+    util-linux \
```
Comment (lumag) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: I thought you said you need only `chrt` util.

File: recipes-test/images/initramfs-test-image.bb (hunk 4 above)
Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb:22]: yes, cgps needs the same.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend (hunk 5 above)
Comment (bhupesh-sharma) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:8]: qcom uses some user-sapce apps based on libgpiod for gps enablement. Idea is to let them run the same, if they cross compile and add it to the initramfs_xxx.cpio.gz

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend (hunk 6 above)
Comment (bhupesh-sharma) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: Strange that the pull request doesn't show the updated commit log :(
I added the details there:
-> From 'meta-oe' layer:
    - lsof: list open files.
    - ncurses: for enabling cgps like tools on target.
    - rt-test: for 'lmbench', etc.
    - stress-ng: to load and stress a system.

-> From 'openembedded-core' layer:
    - crash: for analyzing a vmcore obtained from crash'ed kernel.
    - libgpiod-tools / libgpiod: for gpio set/get, etc.
    - mbw: for benchmarking.
    - util-linux-chrt: for 'chrt'

Comment (bhupesh-sharma) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: util-linux: for 'gdisk', 'taskset', etc.

Comment (bhupesh-sharma) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: May be I need to update the commit log again :(

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend (hunk 2 above)
Comment (ndechesne) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:2]: unless github is playing with me, I still see the same issue in v2.

File: recipes-test/images/initramfs-test-image.bb [hunk 7]
```diff
@@ -17,10 +17,16 @@ PACKAGE_INSTALL += " \
...
+    ncurses-terminfo-base \
     pciutils \
     pd-mapper \
     qrtr \
     rmtfs \
+    rt-tests \
```
Comment (ndechesne) [inline recipes-test/images/initramfs-test-image.bb:28]: do you really need rt-tests, and not just lmbench? I don't even see how lmbench is pulled into the image

File: recipes-test/images/initramfs-test-image.bb (hunk 4 above)
Comment (lumag) [reply recipes-test/images/initramfs-test-image.bb:22]: Interesting. I'm using cgps with just ncurses-terminfo-base, no need for ncurses of full ncurses-terminfo.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend (hunk 6 above)
Comment (lumag) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: I'd prefer to see individual packages instead of full `util-linux` suite here. Not to mention that `gdisk` comes from `gptfdisk` package, rather than `util-linux`.

File: recipes-test/images/initramfs-test-image.bb (hunk 7 above)
Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb:28]: Yes, they are needed for cyclictest as well, which are used for running low-latency / rt checks.

File: recipes-test/images/initramfs-test-image.bb (hunk 4 above)
Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb:22]: Strange. May be you are using some other packages added to your receipe as well.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend (hunk 6 above)
Comment (bhupesh-sharma) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: There are several of those. It would be easier to add the full package rather than individual ones and then realize that some default busybox utility doesn't work well whereas the 'util-linux' one does (I have already landed into such situations before). Anyways the overall size increment is not too much for the whole 'util-linux' package.

Comment (lumag) [reply dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-image.bbappend:12]: My opinion would be that we should still keep it small. Getting test utils is fine. Getting individual software packages too. But getting a bunch of software 'just in case' generally is not from my point of view.

**Comment by ndechesne** [comment]
we've discussed with @bhupesh-sharma , and agreed to introduce initramfs-test-full-image with 'more' utilities. So that we have:
1. initramfs-tiny
2. initramfs-test
3. initramfs-test-full 
Where we incrementally add more packages, so that users can pick the appropriate image they need.

File: recipes-test/images/initramfs-test-image.bb (hunk 7 above)
Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb]: As @ndechesne mentioned, I have pushed 'initramfs-test-full' related changes in the latest PR version.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend [hunk 8]
```diff
@@ -0,0 +1,14 @@
+PACKAGE_INSTALL += " \
+    coreutils \
```
Comment (ndechesne) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend:2]: coreutils is in OE-core, it should not be in the bbapend file. only packages which are built from recipes from meta-oe should be here.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    crash \
+    cryptsetup \
+    devmem2 \
+    iozone3 \
+    kexec \
```
Comment (ndechesne) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend:7]: kexec is OE-core, no?

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    libgpiod \
+    libgpiod-tools \
+    makedumpfile \
+    mbw \
```
Comment (ndechesne) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend:11]: mbw is in master only, so it will fail when backporting in dunfell. we need the recipe in meta-backports (https://git.linaro.org/openembedded/meta-backports.git/). so the backport won't be trivial.

File: dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    util-linux \
```
Comment (ndechesne) [inline dynamic-layers/openembedded-layer/recipes-test/images/initramfs-test-full-image.bbappend:12]: oe-core, not meta-oe

File: recipes-test/images/initramfs-test-image.bb (hunk 4 above)
Comment (lumag) [reply recipes-test/images/initramfs-test-image.bb]: Nope. Please recheck w/o ncurses-terminfo. TERM=linux is included in the ncurses-terminfo-base.

Comment (bhupesh-sharma) [reply recipes-test/images/initramfs-test-image.bb]: I tried it earlier, but it did not work for me. Hmm.. Let me try again But anyways we have these in the 'initramfs-test-full-image.bb ' instead of 'initramfs-test-image.bb', so increase size shouldn't be an issue.

**Comment by ndechesne** [comment]
I am going to merge as is, since we really need it merged now. we will do further improvements later (ncurses and some cleanup such as building on top of initramfs-test instead of tiny).
